server/src/internalClusterTest/java/org/elasticsearch/search/query/SearchQueryIT.java

```diff
@@ -1626,6 +1627,33 @@ public void testRangeQueryWithLocaleMapping() throws Exception {
         assertHitCount(searchResponse, 2L);
     }
 
+    public void testDistanceFeatureQuery() {
```

Comment:
It would be nice to rely on unit tests like `DistanceFeatureQueryBuilderTests` or a REST test instead of an integration test. We try to avoid new `ESIntegTestCase` cases unless a full cluster is really needed.

Reply:
@jtibshirani Thanks for the suggestion. I was trying to make this test in REST yml tests, but yml tests don't have nice comparison of float scores (you can't set up delta, and also there are errors with double -> float conversions).

I also considered adding this test in `DistanceFeatureQueryBuilderTests`, but then I need to work on the Lucene level there. But what I wanted to test is how ES parses this query and what scores are returned.  WDYT?

Reply:
> yml tests don't have nice comparison of float scores 

Got it, that's good to know.

> But what I wanted to test is how ES parses this query and what scores are returned. 

I guess we already test in `DistanceFeatureQueryBuilderTests` that the query is parsed correctly, since we check `LatLonPointDistanceFeatureQuery` has a boost of 1.0f. I also don't see a convenient way to test query scoring in a unit test.

Reply:
@jtibshirani  

>I guess we already test in DistanceFeatureQueryBuilderTests that the query is parsed correctly, since we check LatLonPointDistanceFeatureQuery has a boost of 1.0f.

Indeed, the tests in DistanceFeatureQueryBuilderTests should be enough for this PR. I've removed a test from `SearchQueryIT.java` in 7cbe160ec5a32.  We can introduce this test on scores later, when we need to test the correct scores.